src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -717,11 +708,14 @@ private enum StorageAccessType
     {
         // Console.WriteLine($"Accessing {storageCell} {storageAccessType}");
 
-        long storageWitnessGas = vmState.Env.Witness.AccessForStorage(storageCell.Address, storageCell.Index,
-            storageAccessType == StorageAccessType.SSTORE);
-        if (!UpdateGas(storageWitnessGas, ref gasAvailable)) return false;
+        var gasBefore = gasAvailable;
+        if (!vmState.Env.Witness.AccessForStorage(storageCell.Address, storageCell.Index,
+                ref gasAvailable, storageAccessType == StorageAccessType.SSTORE))
+        {
+            return false;
+        }
 
-        bool witnessGasCharged = storageWitnessGas != 0;
+        bool witnessGasCharged = gasBefore - gasAvailable != 0;
```

Comment:
use a consistent way to do this
either use this  `witnessGasCharged = gasBefore != gasAvailable` or ` witnessGasCharged = gasBefore - gasAvailable != 0`, but use the same everywhere